src/snowflake/snowpark/table.py

```diff
@@ -485,6 +506,34 @@ def update(
                 condition is not None
             ), "condition should also be provided if source is provided"
 
+        kwargs = {}
+        stmt = None
+        if _emit_ast:
+            stmt = self._session._ast_batch.assign()
+            ast = with_src_position(stmt.expr.sp_table_update, stmt)
+            debug_check_missing_ast(self._ast_id, self)
+            ast.id.bitfield1 = self._ast_id
+            if assignments is not None:
+                for k, v in assignments.items():
+                    t = ast.assignments.add()
+                    t._1 = k
+                    build_expr_from_snowpark_column_or_python_val(t._2, v)
+            if condition is not None:
+                build_expr_from_snowpark_column(ast.condition, condition)
+            if source is not None:
+                source._set_ast_ref(ast.source)
+            if statement_params is not None:
+                for k, v in statement_params.items():
+                    t = ast.statement_params.add()
+                    t._1 = k
+                    t._2 = v
```

Comment:
These few lines repeat several times in this module, and I bet the pattern shows up in a few other modules, too. Time for a `build_expr_from_dict`?